src/api/v1/db/oracledb/ingredients-dao.js

```diff
@@ -196,4 +196,24 @@ const postIngredient = async (body) => {
   }
 };
 
-export { getIngredients, postIngredient };
+/**
+ * Get a single ingredient with ID `id` from the database
+ *
+ * @param {string} id
+ * @returns {Promise<object>} the ingredient with ID `id`
+ */
+const getIngredientById = async (id) => {
+  const query = getIngredientsQuery('ID = :id');
+  const bindParams = { id };
+  const connection = await getConnection();
+  try {
+    const { rows } = await connection.execute(query, bindParams);
+    if (rows.length === 1) return serializeIngredient(rows[0], `ingredients/${id}`);
+    if (rows.length > 1) throw new Error("database return shouldn't have multiple rows");
+    return null;
```

Comment:
I would rewrite this as
```
if (rows.length > 1) throw new Error("database return shouldn't have multiple rows");
if (rows.length === 1) {
  return serializeIngredient(rows[0], `ingredients/${id}`);
} else {
  return null;
}
```
This is makes the method more clear on what values it can return. You should strive to only have 1 return statement (same goes for exit conditions in loops) because it makes it clear where the method ends. We don't enforce this though and in this case I think it is fine to have 2.

Reply:
I've [read the opposite](https://blog.timoxley.com/post/47041269194/avoid-else-return-early) in some cases but don't really have a strong take on it. I like how your suggested implementation makes the control flow really clear in this case though, and there's not really a risk that we'll end up even further nesting the `if`/`else` statements.

Reply:
Hmm, eslint is marking that [as a linter error based on this rule](https://eslint.org/docs/rules/no-else-return). Given that, I think my inclination would be to do this:

```js
    if (rows.length > 1) throw new Error("database return shouldn't have multiple values");
    if (rows.length === 0) return null;
    return serializeDough(rows[0], `doughs/${id}`)
```

So it's checking the two "error" cases and then the return at the bottom is the "default" case.

Reply:
Yeah the single point of exit is debatable. I lean toward having one return but its good to recognize points were multiple returns make more sense. The else was a mistake I like your solution.